symphony/bdk/core/config/model/bdk_config.py

```diff
@@ -26,10 +27,18 @@ def __init__(self, **config):
         self.ssl = BdkSslConfig(config.get("ssl"))
         self.app = BdkAppConfig(config.get("app"))
         self.datafeed = BdkDatafeedConfig(config.get("datafeed"))
+        self.retry = BdkRetryConfig(config.get("retry"))
 
     def is_bot_configured(self) -> bool:
         """
 
         :return: true if bot service account is specified in the configuration.
         """
         return self.bot.username is not None
+
+    def get_datafeed_retry_config(self) -> dict:
+        """
+
+        :return: Datafeed retry config if present, Bdk retry config otherwise
+        """
+        return self.retry if self.datafeed.retry is None else self.datafeed.retry
```

Comment:
`self.datafeed.retry` is never `None` (see constructor of `BdkDatafeedConfig` below). This means we are never going to take the global retry config.